src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java

```diff
@@ -337,4 +339,39 @@ public Subscriptions getSubscriptions() {
     private boolean stringToBoolean(String value) {
         return StringUtils.equalsIgnoreCase("yes", value);
     }
-}
+
+    public SscsDocument getLatestDocumentForDocumentType(DocumentType documentType) {
+        if (getSscsDocument() != null && getSscsDocument().size() > 0) {
+            Stream<SscsDocument> filteredDocs = getSscsDocument().stream().filter(f -> f.getValue().getDocumentType().equals(documentType.getValue()));
+
+            List<SscsDocument> docs = new ArrayList<>(filteredDocs.collect(Collectors.toList()));
+
+            docs = sortDocumentsByDateAdded(docs);
+
+            if (docs.size() > 0) {
+                return docs.get(0);
+            }
+        }
+        return null;
+    }
+
+    public List<SscsDocument> sortDocumentsByDateAdded(List<SscsDocument> docs) {
```

Comment:
why not add SSCSDocuments implement Comparable? So you then don't need this.

Reply:
done